src/frontend/context/SøknadContext.ts

```diff
@@ -148,7 +148,7 @@ const [SøknadProvider, useSøknad] = createUseContext(
                 // Ny behandling er lastet som ikke har fullført søknad-steget.
                 tilbakestillSøknad();
             }
-        }, [åpenBehandling]);
+        }, [JSON.stringify(åpenBehandling.søknadsgrunnlag)]);
```

Comment:
Hvorfor må vi stringifye denne?

Reply:
Dette ble egentlig løst gjennom empirisk testing - og medfører at skjema ikke blir tilbakestilt når man legger til/fjerner manuelle brevmottakere og ellers fungerer som tidligere. Fikk tips av Tor..

Alternativt kan jeg forsøke å løse det tilsvarende svaret til Tholle her om du synes det virker som en bedre løsning totalt sett (er vel forklaring også der): 
https://stackoverflow.com/questions/55808749/use-object-in-useeffect-2nd-param-without-having-to-stringify-it-to-json/55808960#55808960
Selv om det visstnok anses mer "korrekt" synes jeg det virker litt "bloated" med to ekstra useRef og mye ekstra kode for å oppnå det samme..?

Reply:
Hmm ok. Slik jeg tolker det ønsker vi å kjøre koden som er i useEffecten når søknadsgrunnlaget lagret på åpenBehandling endrer seg. Når endrer søknadsgrunnlaget seg egentlig på denne siden? 

Slik jeg ser koden har vært før så gjorde vi dette hver gang åpenBehandling endrer seg. I hvilken tilfeller var det, knekker det noe om vi endrer det?

Reply:
Tidligere var parameter til useeffect hele "åpenBehandling". 
I åpenBehandling inngår manuelleBrevmottakere - som medfører at useEffecten trigges hver gang manuelle brevmottakere legges/til fjernes - og endrede data i skjema som ikke er lagret dermed tilbakestilles (rerender).
Jeg antar at dette ikke er ønskelig - så dette er ment som en fiks for dette.

Om man videre endrer til kun "åpenBehandling.søknadsgrunnlag" som først forsøkt - virker det som objekt-referansen til åpenbehandling endres ved legg til/fjern brevmottaker, noe som igjen oppdaterer åpenBehandling.søknadsgrunnlag og fortsatt gir samme "feil" at ulagrede data i skjema forsvinner.

Ved å benytte JSON-strinfigy sammenlignes i stedet selve objektets innhold som medfører at det ikke har endret seg - ettersom innoldet og ikke referansen sammenlignes - og dataene forblir i skjema. 

Så spørsmåler er om dette virker greit eller om det er mer ønskelig med en mer omfattende løsning som skissert av Tholle i stackoverflow link over. Hvorfor det ev anses mer korrekt kan jeg ikke forstå. 😊 Mye mer kode.

Reply:
Ja, forstår at den har forskjellig referanse. Jeg lurer på i hvilke tilfeller søknadsgrunnlaget endrer seg slik at man ønsker å gjøre det som skjer i useEffecten? eventuelt slik det var før; i hvilke andre tilfeller (annet enn endret brevmottaker) gjør at åpenBehandling endrer seg og man ønsker å gjøre det som skjer i useEffecten? 

Klarer ikke helt å se at man ønsker å tilbakestille søknaden med data fra backend annet enn når man laster siden? I såfall trenger vi ikke dependency array i det hele tatt 🤔

Reply:
Slik jeg forstår det er hensikten å oppdatere skjemaverdiene når søknadsgrunnlaget endrer seg.
Mulig du har rett - men må nesten spørre tilbake da jeg er veldig usikker - ganske uoversiktlig det hele..

Kunne du gitt din vurdering på om det du foreslår er en mulighet - å bare fjerne dependency array helt - noe som vel innebærer en større endring - og at useEffect blir kalt hyppigere..?

Det var som sagt Tor som kom med dette innspillet - noe som jeg gjorde at jeg følte meg relativt trygg på at endringen var sikker å foreta.

Dersom du føler deg like usikker som meg på dette - særlig å fjerne dependency array helt  - hvem andre kan man forhøre seg med for å føle seg sikrere..? 😊